avro/src/bigdecimal.rs

```diff
@@ -50,27 +51,27 @@ pub(crate) fn deserialize_big_decimal(bytes: &Vec<u8>) -> AvroResult<BigDecimal>
     let mut bytes: &[u8] = bytes.as_slice();
     let mut big_decimal_buffer = match decode_len(&mut bytes) {
         Ok(size) => vec![0u8; size],
-        Err(err) => return Err(Error::BigDecimalLen(Box::new(err))),
+        Err(err) => return Err(Details::BigDecimalLen(Box::new(err)).into()),
```

Comment:
We can drop the usage of Box here, right ?

Reply:
No, that would make the Error struct infinitely large (Error -> Details -> Error -> Details ...)